### ISSUE TITLE
Examples drift: debug_cartpole.rs has stale Result-returning reset()/step() syntax

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#25]
## Symptom

`cargo check --features training --examples` reports ~9 errors all in `examples/games/cartpole/debug_cartpole.rs`. The example uses stale `?`-propagation on `env.reset()?` and `env.step(action)?`, but the current `Environment` trait (`src/env/mod.rs:9-30`) no longer returns `Result` from those methods.

This is a different drift class than the one fixed by #18 (`TrainingMetadata` / `SpaceInfo` / `buffer.add`) — the `train_cartpole*.rs` files don't have this issue, only `debug_cartpole.rs` does.

## Reproduction

```bash
cargo check --features training --example debug_cartpole
```

## Acceptance

- `cargo check --features training --examples` succeeds with zero errors.
- `cargo build --features training --example debug_cartpole` succeeds.
- The example's diagnostic output is preserved (don't lose behavior, just retype the call sites).

## Discovery

Flagged by the curator and builder for #18 ([curator comment](https://github.com/rjwalters/thrust/issues/18#issuecomment-4455226132), and noted in PR #23). Explicitly scoped out of #18 because the breakage class is distinct.

**Comment by rjwalters** [comment]
## Curator enhancement

### Verified error count and locations

I could not run `cargo check` end-to-end in this environment (libtorch not installed locally — torch-sys fails its build script), so I verified the breakage by source-level inspection against the authoritative trait. The original report's \"~9 errors\" was an overestimate. There are **exactly 2 lines** in `examples/games/cartpole/debug_cartpole.rs` that use the stale `Result`-returning shape:

| Line | Current (broken) | Why it fails |
|------|------------------|--------------|
| `examples/games/cartpole/debug_cartpole.rs:18` | `let mut obs = env.reset()?;` | `Environment::reset` now returns `()` (`src/env/mod.rs:11`), so (a) the `?` is on a non-`Try` type and (b) `reset` no longer hands back the observation |
| `examples/games/cartpole/debug_cartpole.rs:39` | `let result = env.step(action)?;` | `Environment::step` now returns `StepResult` directly, not `Result<StepResult>` (`src/env/mod.rs:17`) |

Each of those typically surfaces as a small cluster of diagnostics from rustc (the `?` itself, the inferred type mismatch on `obs`, follow-on errors where `obs` is consumed as `&Vec<f32>` into `Tensor::from_slice`), which is plausibly where the \"~9\" came from in the original report. The root causes are the two lines above.

### Scope check: no other drift in this file

I scanned `debug_cartpole.rs` for the full set of drift patterns that have bitten sibling examples in #16, #18, #21:

- `SpaceInfo.dtype` / `SpaceType::Continuous` — not used in this file. The example never reads the space type.
- `TrainingMetadata { hyperparameters, ... }` (the #18 drift) — not used; this is a diagnostic, not a training/export example.
- `RolloutBuffer::add` signature change (`#23`) — not used.
- `batch.log_probs` / `batch.values` rename — not used.

The remaining `?` operators in the file are **legitimate** and should be left alone:
- `examples/games/cartpole/debug_cartpole.rs:60` — `Vec::try_from(test_actions)?` — `tch` `Vec::try_from(Tensor)` returns `Result<Vec<T>, TchError>`. Same pattern is live and correct in `examples/games/cartpole/train_cartpole_best.rs:134-136` and `train_cartpole_modern.rs:129-131`.
- `examples/games/cartpole/debug_cartpole.rs:61` — same reason.
- `examples/games/cartpole/debug_cartpole.rs:80` — `entropy.try_into()?` — `f64::try_from(Tensor)` returns `Result<f64, TchError>`. Legitimate.

So the fix is genuinely narrow: only the two `?` removals plus reconstructing the initial observation, since `reset` no longer hands it back.

### Authoritative current shape

From `src/env/mod.rs:9-30`:
```rust
pub trait Environment {
    fn reset(&mut self);
    fn get_observation(&self) -> Vec<f32>;
    fn step(&mut self, action: i64) -> StepResult;
    // ...
}
```

The migration pattern is identical to what commit `2027f61` (PR #16, closing #7) applied to the `#[cfg(test)]` block in `src/env/games/cartpole.rs`. From that diff:

> *\"restructure `let obs = env.reset()` to `env.reset(); let obs = env.get_observation();` since `reset` no longer returns the observation.\"*

You can see the live shape of that pattern at `src/env/games/cartpole.rs:249-253` (`test_cartpole_reset`) and at `src/env/games/cartpole.rs:265-268` (`test_cartpole_step`, which simply calls `env.step(1)` with no `?`).

### Implementation guidance

Two mechanical edits in `examples/games/cartpole/debug_cartpole.rs`:

**Edit 1 — line 18** (currently `let mut obs = env.reset()?;`):
```rust
env.reset();
let mut obs = env.get_observation();
```
This keeps `obs: Vec<f32>` so the downstream `Tensor::from_slice(&obs).reshape([1, 4])` on line 25 continues to work unchanged. Note: `let mut` is still required because line 51 reassigns `obs = result.observation;` inside the loop.

**Edit 2 — line 39** (currently `let result = env.step(action)?;`):
```rust
let result = env.step(action);
```
`result.reward` / `result.terminated` / `result.truncated` / `result.observation` accesses on lines 41-51 remain valid — `StepResult` exposes those fields directly (`src/env/mod.rs:33-49`).

The `main() -> Result<()>` signature stays — it's still needed for lines 60, 61, 80 (the legitimate `?` operators) and the trailing `Ok(())` on line 91.

### Acceptance criteria

- [ ] `cargo check --features training --example debug_cartpole` succeeds with zero errors.
- [ ] `cargo check --features training --examples` succeeds with zero errors (regression guard for the full example matrix).
- [ ] `cargo build --features training --example debug_cartpole` succeeds.
- [ ] Diagnostic output is preserved verbatim — only the two `?`-on-`reset`/`step` call sites change shape; print statements, the policy entropy block, and action-distribution sampling are untouched.
- [ ] No new clippy warnings introduced (`cargo clippy --features training --example debug_cartpole -- -D warnings`).

### Test plan

- Run `cargo check --features training --example debug_cartpole` — expect clean build.
- Run `cargo build --features training --example debug_cartpole` — expect clean link.
- Run `cargo run --features training --example debug_cartpole` — expect output to include \"🔍 Debugging CartPole training...\", an \"Initial observation: [...]\" line with four small floats, 1-10 step lines (the episode may terminate early — that is expected for an untrained policy), the \"📊 Checking policy distribution...\" block, action counts that sum to 100, and one of the entropy verdict lines (\"⚠️  Policy is nearly deterministic\" or \"✅ Policy has reasonable entropy\"). With a freshly initialized MLP the entropy will be near `ln(2) ≈ 0.693`, so the success branch is the expected path.
- Run `cargo check --features training --examples` to confirm no other example regressed.

### Relevant code references

- `src/env/mod.rs:9-30` — current `Environment` trait (non-`Result` `reset` and `step`).
- `src/env/mod.rs:33-49` — `StepResult` struct (fields used by `debug_cartpole.rs`).
- `src/env/games/cartpole.rs:184-216` — `Environment for CartPole` impl confirming the concrete return types.
- `src/env/games/cartpole.rs:249-253` and `:265-268` — in-tree examples of the post-migration `reset() / get_observation() / step()` call shape.
- Commit `2027f61` (PR #16) — original migration of the same idiom in the lib test code; the rationale comment there is the canonical reference.
- PR #23 (commit `d43a98f`, closing #18) — fixed neighboring `train_cartpole*.rs` files; explicitly noted that `debug_cartpole.rs` was out of scope and would need this separate issue.

### Estimated complexity

Trivial — two-line surgical change with no logic impact. Suitable as a first-touch task for a builder.
